controllers/products.js

```diff
@@ -117,32 +118,92 @@ module.exports = function(app, config) {
 						query+= " limit "+page+" , "+limit;
 				
 					app.models.products
-					.query(query, function(err, cont) {
+					.query(query, function(err, products) {
 						if (err) return autoCallback(err);
-						return autoCallback(null, cont);
+						return autoCallback(null, products);
 					});
 				}]
 			},
 			function(err, results) {
 				if (err) return next(err);
-				var products = _.map(results.getRecords, function(productData) {
+				var productFData = {},
+				productPrefix = results.getProductPrifix;
+				var quantityInfo = {};
+					var priceInformationInfo = {};
+				async.forEachOf(results.getRecords, function(productData, key, eachCallBack) {
 					// add the quantity info in the response data 
-					var quantityInfo = {};
-					modulesConfig.moduleAttributes.Products.quantity.forEach( function(attr) {
-						quantityInfo[attr]= productData[attr];
-					}); 
-					productData.quantity = quantityInfo;
 					
-					//add price_information info in to response data
-					var priceInformationInfo = {};
-					modulesConfig.moduleAttributes.Products.price_information.forEach( function(attr) {
-						priceInformationInfo[attr] = productData[attr];
-					});
-					productData.price_information = priceInformationInfo;
+					async.auto({
 					
-					return productData;
-				});
-				
+						setQuantity: function(modifiedCallBack) {
+							modulesConfig.moduleAttributes.Products.quantity.forEach( function(attr) {
+								quantityInfo[attr]= productData[attr];
+							}); 
+							productData.quantity = quantityInfo;
+							return modifiedCallBack();
+						},
+					
+						//add price_information info in to response data
+						setPriceInfo: function(modifiedCallBack){
+							modulesConfig.moduleAttributes.Products.price_information.forEach( function(attr) {
+								priceInformationInfo[attr] = productData[attr];
+							});
+							productData.price_information = priceInformationInfo;
+							return modifiedCallBack();
+						},
+						
+						// set the tax and product number
+						setTaxAndProductNumValues: function(modifiedCallBack) {
+							_.map(productData, function(v, k) {
+									// making sure that the tax_values and shipping_handling_tax_values are parsed into proper format
+								if (k === 'tax_values' || k === 'shipping_handling_tax_values') {
+									v = common.parseTaxData(v);
+									productData[k] = v;
+								}
+								// prefixing the SO number with the SO prefix from setting
+								if (productPrefix && productPrefix.settingData && k === 'product_number') {
+									v= productPrefix.settingData+''+v;
+									productData[k] = v;
+								}
+							});
```

Comment:
@vivekshekar 
- Product has only `tax_value` and there is no other tax associated.
- So check only for `tax_value`
- Product does not have customized product number so we do not need that formatting part
- May be rename the function as `setProductTaxValues`